packages/jsii-dotnet-generator/src/Amazon.JSII.Generator.UnitTests/Class/AbstractClassProxyGeneratorTests.cs

```diff
@@ -0,0 +1,188 @@
+using Amazon.JSII.Generator.Class;
+using Amazon.JSII.JsonModel.Spec;
+using Xunit;
+
+namespace Amazon.JSII.Generator.UnitTests.Class
+{
+    public class AbstractClassProxyGeneratorTests : GeneratorTestBase
+    {
+        private const string Prefix = nameof(Generator) + "." + nameof(AbstractClassProxyGenerator) + ".";
+
+        private string Render(ClassType classType)
+        {
+            Symbols.MapTypeToPackage("myFqn", classType.Assembly);
+            Symbols.MapNamespace(classType.QualifiedNamespace, "MyNamespace");
+            Symbols.MapTypeName("myFqn", "MyClass", TypeKind.Class);
+
+            var generator = new AbstractClassProxyGenerator(classType.Assembly, classType, Symbols, Namespaces);
+
+            var syntaxTree = generator.CreateSyntaxTree();
+            return syntaxTree.ToString();
+        }
+
+        [Fact(DisplayName = Prefix + nameof(IncludesAttribute))]
+        public void IncludesAttribute()
+        {
+            var classType = new ClassType
+            (
+                "myFqn",
```

Comment:
I've found that these tests are much less brittle when I use named parameters for all api/spec types. This has several benefits:

1. Less of a false positive or false negative when the api/spec changes.
2. Less work to update tests when the spec changes.
3. Robust against changes to the ordering of parameters, especially adjacent parameters of the same type.

Reply:
Sounds good, but this is a good bit of work somewhat unrelated to this change. I will create a new issue to track this. https://github.com/awslabs/jsii/issues/242